pkg/collector/reconcile/configmap_test.go

```diff
@@ -0,0 +1,95 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reconcile
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestDesiredConfigMap(t *testing.T) {
+	t.Run("should return expected config map", func(t *testing.T) {
+		expected := configMap("test-collector")
+		actual := desiredConfigMap(context.Background(), params())
+		assert.Equal(t, expected, actual)
+	})
+
+}
+
+func TestExpectedConfigMap(t *testing.T) {
+
+	cm := configMap("test-collector")
```

Comment:
You can call the desiredConfigMap here, as you are testing its correctness in another test.

Reply:
Done